src/backends/r1cs/mod.rs

```diff
@@ -126,40 +85,44 @@ where
     }
 
     /// Create a linear combination to represent constant one.
-    fn one() -> Self {
+    fn one(span: Span) -> Self {
         LinearCombination {
             terms: HashMap::new(),
             constant: F::one(),
+            span,
         }
     }
 
     /// Create a linear combination to represent constant zero.
-    fn zero() -> Self {
+    fn zero(span: Span) -> Self {
         LinearCombination {
             terms: HashMap::new(),
             constant: F::zero(),
+            span,
```

Comment:
it's always hard for me to think of "what should have a span"? I think the real answer is "this needs a span because otherwise the error I'm seeing right now is not great"

Reply:
I think the span here would be helpful to debug the intermediate linear combinations (non constraints).